packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -175,14 +175,14 @@ export class MediaPlaceholder extends Component {
 			const isVideo = isOneType && 'video' === allowedTypes[ 0 ];
 
 			if ( instructions === undefined && mediaUpload ) {
-				instructions = __( 'Drag a media file, upload a new one or select a file from your library.' );
+				instructions = __( 'Upload a media file or pick one from your media library.' );
```

Comment:
This one doesn't contain the "or add one with a URL" part, because the media & text block doesn't include that option.